### PR TITLE
UID2-2340 Add test pipeline for Azure

### PR DESCRIPTION
**Pull request by cYKatherine**
Please review in order of:

1. https://github.com/IABTechLab/uid2-operator/pull/329
2. https://github.com/IABTechLab/uid2-shared-actions/pull/50
3. https://github.com/IABTechLab/uid2-operator/pull/328

I’ve tested it working: 
- As an individual E2E task: https://github.com/IABTechLab/uid2-operator/actions/runs/7536390454/job/20513736679
- As a part of the publishing azure operator pipeline: https://github.com/IABTechLab/uid2-operator/actions/runs/7535436593

### CODE REVIEW

File: .github/workflows/shared-run-e2e-tests.yaml [hunk 1]
```diff
@@ -157,6 +157,14 @@ jobs:
           gcp_service_account: ${{ inputs.gcp_service_account }}
           gcp_workload_identity_provider_id: ${{ inputs.gcp_workload_identity_provider_id }}
 
+      - name: Prepare Azure metadata
+        id: prepare_azure_metadata
+        if: ${{ inputs.operator_type == 'azure'}}
```
Comment (gmsdelmundo) [inline .github/workflows/shared-run-e2e-tests.yaml:162]: Space at the end

File: .github/workflows/shared-run-e2e-tests.yaml [hunk 2]
```diff
@@ -188,6 +196,19 @@ jobs:
           image_hash: ${{ steps.prepare_gcp_metadata.outputs.image_hash }}
           operator_key: ${{ steps.prepare_gcp_metadata.outputs.operator_key }}
 
+      - name: Start Azure private operator
+        id: start_azure_private_operator
+        if: ${{ inputs.operator_type == 'azure'}}
```
Comment (gmsdelmundo) [inline .github/workflows/shared-run-e2e-tests.yaml:201]: Space at the end

File: actions/prepare_azure_metadata/action.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Prepare GCP Metadata
```
Comment (gmsdelmundo) [inline actions/prepare_azure_metadata/action.yaml:1]: I think you need a `description` here as well

File: actions/prepare_azure_metadata/action.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+inputs:
+  operator_image_version:
+    description: 'The version of UID2 operator image'
+    type: string
```
Comment (gmsdelmundo) [inline actions/prepare_azure_metadata/action.yaml:5]: `type: string` isn't needed

File: actions/prepare_azure_metadata/action.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+name: Prepare GCP Metadata
+inputs:
+  operator_image_version:
+    description: 'The version of UID2 operator image'
```
Comment (gmsdelmundo) [inline actions/prepare_azure_metadata/action.yaml:4]: No need for single quotes in `description` and `default`

**Review by gmsdelmundo — APPROVED** [review]

File: actions/start_azure_private_operator/action.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+name: Start Azure Private Operator
+description: Start the Azure private operator with the core and optout instances
+inputs:
+  bore_url_core:
+    description: The bore url for core service
+    required: true
+  bore_url_optout:
+    description: The bore url for optout service
+    required: true
+  bore_url_localstack:
```
Comment (thomasm-ttd) [inline actions/start_azure_private_operator/action.yaml:10]: Is this used?

File: actions/start_azure_private_operator/action.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    description: The bore url for localstack service
+    required: true
+  output_template_file:
+    description: The output template file
+    required: true
+  output_parameters_file:
+    description: The output parameters file
+    required: true
+  operator_key:
```
Comment (thomasm-ttd) [inline actions/start_azure_private_operator/action.yaml:19]: Is this used?

**Review by thomasm-ttd — APPROVED** [review]

**Pull request merged**